atomic_reactor/plugins/pre_koji.py

```diff
@@ -36,15 +38,33 @@ def __init__(self, tasker, workflow, target, hub, root, proxy=None):
         self.pathinfo = koji.PathInfo(topdir=root)
         self.proxy = proxy
 
+    def _check_target_and_tag(self):
```

Comment:
It definitely has a way to pull the component from the Dockerfile as it's currently done to check for existing nvr.
The benefit of having the check in k-cb is that the build will fail earlier.